features/glance.feature

```diff
@@ -0,0 +1,39 @@
+@glance
+Feature: Glance
+  As an administrator
+  I want to make sure that Glance component is working correctly
+
+  @create_kvm_image
+  Scenario: Creation of KVM image
+    Given KVM source image exists
+    And glance image does not exist
+    When I create new glance image based on jeos1
+    Then the image is listed as active
+    And this image has non-empty ID
+    And its ID can be used to show the image info
+    When I delete the image
```

Comment:
> @aspiers I don't agree that removing of an image is different from testing.

I didn't say it was - I said *cleanup* is different from actual testing.

> Everything what is being done should be captured in the scenario text.

But that's not really true - there is a TON of setup done which is not captured by the scenario text (setup of the nodes, installation of SUSE Cloud, creation and application of Crowbar barclamp proposals...).

> I consider that as best practice, otherwise we end up with a lot of action hidden before the our eyes when reading the feature and spenario specifications.

As per above we already have a lot of stuff hidden.  But that's fine.  The scenario is only supposed to test one thing.

> It's also not practical to put that into an another scenario because every scenario must be independent, in this case we would create dependency by calling one scenario from the other.  In case the latter fails, you end up looking for the reason why this particular scenario was run. It is clear e.g. for a Background, not for other cases.

No, I think you misunderstand me.  I am definitely not suggesting to create a dependency (and even if I was, I'm not sure if Cucumber actually supports doing that).

> I would not put constraints on what is actual testing and what not. If the scenario fails while the image is being deleted, it's straight forward to see it, more effort is actually not needed compared to a separate scenario having failed, the reason will be the same.

I disagree - the problem is that it breaks the test suite report, because if something breaks with image deletion, any scenario which has image deletion at the end will be reported as failing.  That's bad, because a) it makes the situation look worse than it actually is, and b) extra effort is required to check all failing scenarios and ascertain that they're all only due to a single issue with image deletion.

In contrast, if image deletion is moved to a teardown hook, and tested explicitly via an independent image deletion scenario, then if it fails, you only get a single failure in the report, and you instantly know without any further examination that the failure is directly related to image deletion.

> Additionally, I would like to avoid using hooks as it could end up messy

Why?

> and rather keep the scenarios' as simple as possible

Yes I agree - my proposal makes them simpler than they currently are.